### ISSUE TITLE
Allow use of Alt Space instead of Shift Space keyboard shortcut

### ISSUE THREAD

**Issue by skyeagle7** [sumatrapdfreader/sumatrapdf#957]
I like the scroll by screen keyboard shortcut which is "space" for screen down and "shift space" for screen up.
However I would like to see "alt space" as an option instead of "shift space" for better efficiency.  The alt key on most keyboards is right next to the space bar, so it would make scrolling up or down faster and more efficient.  The shift key is just far enough away to make it a bit more troublesome and takes extra time to use.

I did not see that "shift space" was used anywhere else.

I had hoped there might be a command line option for this (key remapping), but did not see one.

Anyway, please consider this for a future enhancement.

I do find the product very useful for reading PDF and other documents.  The fact that you can set the default to page width for all documents is a time saver, I like the page count and the fact the "home" and "end" key do work.  I tried several other PDF products but they all had some drawbacks when it comes to reading long documents.

**Comment by SumatraPeter** [comment]
The problem with this is that Alt+Space is the universal Windows shortcut for bringing up any app's System menu.

For example in Firefox:

![1](https://user-images.githubusercontent.com/15380332/34691364-5d60e55a-f4e2-11e7-9ae8-19cff3cc730f.png)

Similarly in Sumatra:

![2](https://user-images.githubusercontent.com/15380332/34691374-651539d6-f4e2-11e7-9f7e-99c55b4638bd.png)

Even if the shortcut can be overridden by an app, it is definitely _not_ a good idea to do so.

**Comment by sterbalr** [comment]
Do you have a favorite list of windows shortcuts?

ctrl+c copy
ctrl+x cut
ctrl+v paste
alt+F4 Close
F1 Help
...

Maybe...
F2 Rename
...

which other ones?

**Comment by SumatraPeter** [comment]
@sterbalr: Who are you directing your query towards? Anyway, perhaps you'll find an official list like [this one](https://support.microsoft.com/en-us/help/12445/windows-keyboard-shortcuts) useful.

**Comment by sterbalr** [comment]
I posted a list of the ones from the menu on this wiki page:

https://github.com/sumatrapdfreader/sumatrapdf/wiki/Keyboard-Shortcuts

It would be nice if someone could review the list and add additional ones.

Thanks!

**Comment by SumatraPeter** [comment]
@sterbalr: Oh, so by "favorite list of windows shortcuts" you actually meant keyboard shortcuts in Sumatra specifically? Why waste time unnecessarily creating a WIki page that essentially duplicates the very first section of [Sumatra's manual](http://www.sumatrapdfreader.org/manual.html)?

**Comment by sterbalr** [comment]
Feel free to remove the text and put a link to authoritative documentation.

The issue came up because someone asked why they couldn’t assign alt+space to a keyboard shortcut.

I’ve been working with wikis for about 5 years and I’ve found that over documenting is helpful in that context.

Thanks

**Comment by skyeagle7** [comment]
I appreciate all the comments and understand why the "Alt Space" instead of "Shift Space" is not a great idea.  What I will probably do is remap the right "Alt" key to be a "Shift" instead.  I have remapped that key for years on my primary desktop for other purposes and never had an issue with that, so it seems a good alternative for my specific purpose.

The reason this change would be helpful is when reading columnar articles or publications.  Many times the articles are more than a screen full so you have  to read up and down several times to get the full article.  When you have perhaps dozens of articles in a publication, that gets a bit tiresome.

What can I say expect I like to push boundaries and see how to make it easier.

**Comment by SumatraPeter** [comment]
@skyeagle7: I think remapping RAlt to Shift on your own system is a much better solution than Sumatra itself introducing such a major breaking change. There are utilities that will change the key mapping in the Windows registry itself so that you won't even need to have a program constantly running in the background.

**Comment by skyeagle7** [comment]
You are correct.  There are several keyboard remapping programs out there that edit the Windows registry.  I used the one I had before which is KeyTweak.  I was frankly surprised it worked on Windows 10 as I think the program is no longer in active development.  It is freeware and does not require an install.

In any case, I have remapped that rAlt key to "Shift" and it is working great.  If I did not have to read so much columnar reading material, this would not have been such a pesky issue.

Again thanks for all the advice.  Sumatra is my now preferred software for reading documents.  A nice touch is that when you reopen a document, it will open where you left off.
